src/BTVNanoCommissioning/workflows/ctag_Wc_valid_sf.py

```diff
@@ -546,47 +628,9 @@ def process_shift(self, events, shift_name):
             pruned_ev["dr_lep1_softmu"] = shmu.delta_r(ssmu)
             pruned_ev["soft_l_ptratio"] = ssmu.pt / smuon_jet.pt
             pruned_ev["l1_ptratio"] = shmu.pt / smuon_jet.pt
+            pruned_ev["MuonJet_beta"] = smuon_jet.pt / smuon_jet.E
 
-            # Create a list of variables want to store. For objects from the PFNano file, specify as {object}_{variable}, wildcard option only accepted at the end of the string
-            out_branch = np.setdiff1d(
-                np.array(pruned_ev.fields), np.array(events.fields)
-            )
-            out_branch = np.delete(
-                out_branch,
-                np.where(
-                    (out_branch == "SoftMuon")
-                    | (out_branch == "MuonJet")
-                    | (out_branch == "dilep")
-                    | (out_branch == "OtherJets")
-                ),
-            )
-
-            for kin in ["pt", "eta", "phi", "mass", "pfRelIso04_all", "dxy", "dz"]:
-                for obj in [
-                    "Muon",
-                    "Jet",
-                    "SoftMuon",
-                    "MuonJet",
-                    "dilep",
-                    "charge",
-                    "MET",
-                ]:
-                    if "MET" in obj and ("pt" != kin or "phi" != kin):
-                        continue
-                    if (obj != "Muon" and obj != "SoftMuon") and (
-                        "pfRelIso04_all" == kin or "d" in kin
-                    ):
-                        continue
-                    out_branch = np.append(out_branch, [f"{obj}_{kin}"])
-            out_branch = np.append(
-                out_branch, ["Jet_btagDeep*", "Jet_DeepJet*", "PFCands_*", "SV_*"]
-            )
-            # write to root files
-            os.system(f"mkdir -p {self.name}/{dataset}")
-            with uproot.recreate(
-                f"{self.name}/{dataset}/{systematics[0]}_{events.metadata['filename'].split('/')[-1].replace('.root','')}_{int(events.metadata['entrystop']/self.chunksize)}.root"
-            ) as fout:
-                fout["Events"] = uproot_writeable(pruned_ev, include=out_branch)
+            array_writer(self, pruned_ev, events, systematics[0], dataset, isRealData)
```

Comment:
can you put the  `SoftMuon, MuonJet, dilep` as customized variables used in `arraywriter`?